src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -72,10 +74,11 @@ public Dictionary(int capacity, IEqualityComparer<TKey> comparer)
         {
             if (capacity < 0) ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.capacity);
             if (capacity > 0) Initialize(capacity);
-            this.comparer = comparer ?? EqualityComparer<TKey>.Default;
+            this.comparer = comparer;
```

Comment:
I would do :
```C#
if (comparer != EqualityComparer<TKey>.Default)
{
    this.comparer = comparer;
}
```
So when `comparer` param happens to be `EqualityComparer<TKey>.Default` we don't lose the optimization